cmake/OpenCVFindIPP.cmake

```diff
@@ -34,6 +34,11 @@ unset(IPP_VERSION_MAJOR)
 unset(IPP_VERSION_MINOR)
 unset(IPP_VERSION_BUILD)
 
+if (X86 AND UNIX AND NOT APPLE AND NOT ANDROID AND BUILD_SHARED_LIBS)
+    message(STATUS "On 32-bit Linux IPP can not currently be used with dynamic libs. Set BUILD_SHARED_LIBS=OFF")
```

Comment:
@vpisarev, I would see this info as comments in `cmake/OpenCVFindIPP.cmake`, because without these comments it can confuse other users